File: op/1semester/lab3/python/lab3.py
import math

lnPrevious = 0 
ln = 0
e = 0.000001
n=1
x=float(input("Enter value of the \'x\': "))

while True :
    lnPrevious = ln
    ln += (-1)**(n-1) * (((x)**n)/n)
    n+=1
    if abs(ln-lnPrevious) < e:
        break
print(ln)

File: op/1semester/lab3/c++/lab3.cpp
#include<iostream>
#include<math.h>
using namespace std; 



int main(){
    float x, lnPrevious=0,ln=0, e = 0.000001;
    int n=1;
    cout << "Enter value of the \'x\': ";
    cin >> x;
    do {
        lnPrevious = ln;
        ln +=  pow(-1,(n-1)) * float(pow(x,n))/n;
        n++;
    }  while ( abs(ln - lnPrevious) > e);
    cout << ln; 
    return 0;
    }